test/weights.jl

```diff
@@ -36,6 +36,34 @@ weight_funcs = (weights, aweights, fweights, pweights)
     @test sum(sa, wv) === 7.0
 end
 
+# isequal and ==
+
+@testset "isequal" begin
```

Comment:
Should be "isequal and ==", and you can drop the above comment.

It would be good to test this for all weights types. You can do that with a `for` on the testset line, iterating over the type. To test that different types do not compare equal, you can do `if T !== FrequencyWeights`.